pkg/osquery/enrollment_details.go

```diff
@@ -31,22 +37,23 @@ func getEnrollDetails(ctx context.Context, osquerydPath string) (service.Enrollm
 		return details, errors.New("Skipping enrollment details")
 	}
 
+	details.OSPlatform = runtime.GOOS
+	details.OSPlatformLike = runtime.GOOS
+
 	// If the binary doesn't exist, bail out early.
 	if info, err := os.Stat(osquerydPath); os.IsNotExist(err) {
-		return details, fmt.Errorf("no binary at %s", osquerydPath)
+		return details, fmt.Errorf("%w: no binary at %s:", osqueryNotFoundErr{}, osquerydPath)
 	} else if info.IsDir() {
-		return details, fmt.Errorf("%s is a directory", osquerydPath)
+		return details, fmt.Errorf("%w: %s is a directory", osqueryNotFoundErr{}, osquerydPath)
 	} else if err != nil {
-		return details, fmt.Errorf("statting %s: %w", osquerydPath, err)
+		return details, fmt.Errorf("%w: statting %s: %w", osqueryNotFoundErr{}, osquerydPath, err)
 	}
 
 	query := `
 	SELECT
 		osquery_info.version as osquery_version,
 		os_version.build as os_build,
```

Comment:
Do you know if we can get the rest of `os_version` ?

Reply:
I think only if we exec for it. There is `unix.Sysctl("kern.osrelease")`, but it gives darwin kernel version as opposed to macos version.

Reply:
Or we could read `/System/Library/CoreServices/SystemVersion.plist` (and find the equivalent on other platforms)

Reply:
Eh. Maybe we kick that forward